qos-monitor/src/main/java/eu/arrowhead/core/qos/dto/externalMonitor/ExternalMonitorOrchestrationRequestFactory.java

```diff
@@ -0,0 +1,122 @@
+import org.springframework.stereotype.Service;
+
+import eu.arrowhead.common.CommonConstants;
+import eu.arrowhead.common.CoreCommonConstants;
+import eu.arrowhead.common.SSLProperties;
+import eu.arrowhead.common.dto.internal.ServiceDefinitionRequestDTO;
+import eu.arrowhead.common.dto.internal.ServiceInterfaceRequestDTO;
+import eu.arrowhead.common.dto.shared.OrchestrationFlags.Flag;
+import eu.arrowhead.common.dto.shared.OrchestrationFormRequestDTO;
+import eu.arrowhead.common.dto.shared.ServiceQueryFormDTO;
+import eu.arrowhead.common.dto.shared.ServiceSecurityType;
+import eu.arrowhead.common.dto.shared.SystemRequestDTO;
+import eu.arrowhead.core.qos.QosMonitorConstants;
+
+@Service
+public class ExternalMonitorOrchestrationRequestFactory {
+
+	//=================================================================================================
+	// members
+
+	@Autowired
+	protected SSLProperties sslProperties;
+
+	@Value(CoreCommonConstants.$CORE_SYSTEM_NAME)
+	private String coreSystemName;
+
+	@Value(CoreCommonConstants.$SERVER_ADDRESS)
+	private String coreSystemAddress;
+	
+	@Value(CoreCommonConstants.$SERVER_PORT)
+	private int coreSystemPort;
+
+	@Resource(name = CommonConstants.ARROWHEAD_CONTEXT)
+	private Map<String,Object> arrowheadContext;
+
+	private Logger logger = LogManager.getLogger(ExternalMonitorOrchestrationRequestFactory.class);
+
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------
+	public OrchestrationFormRequestDTO createExternalMonitorOrchestrationRequest() {
+		logger.debug("createExternalMonitorOrchestrationRequest started...");
+
+		final SystemRequestDTO requester = createExternalMonitorSystemRequestDTO();
+		final ServiceQueryFormDTO requestedService = createExternalPingMonitorServiceQueryFormDTO();
+
+		final OrchestrationFormRequestDTO orchestrationForm = new OrchestrationFormRequestDTO.Builder( requester )
+				.requestedService(requestedService)
+				.flag(Flag.ENABLE_INTER_CLOUD, false)
+				.flag(Flag.MATCHMAKING, false)
+				.flag(Flag.ENABLE_QOS, false)
+				.flag(Flag.OVERRIDE_STORE, true)
+				.build();
+
+		return orchestrationForm;
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	public SystemRequestDTO createExternalMonitorSystemRequestDTO() {
+		logger.debug("createExternalMonitorSystemRequestDTO started...");
+
+		final PublicKey publicKey = (PublicKey) arrowheadContext.get(CommonConstants.SERVER_PUBLIC_KEY);
+
+		final SystemRequestDTO system = new SystemRequestDTO();
+		system.setSystemName(coreSystemName);
+		system.setAddress(coreSystemAddress);
+		system.setPort(coreSystemPort);
+		system.setMetadata(null);
+		if (sslProperties.isSslEnabled()) {
+			system.setAuthenticationInfo(Base64.getEncoder().encodeToString(publicKey.getEncoded()));
+		}
+
+		return system;
+	}
```

Comment:
should be stored the system request dto in a member. Create an instance for the first request ((lazy initialization)), after that just return the stored value.

Reply:
fixed